.github/workflows/build-image.yml

```diff
@@ -0,0 +1,41 @@
+name: Build Vigilo image and push to ECR
+on:
+   push:
+      branches:
+        - main
+    # Allow running this workflow manually from the Actions tab
+    workflow_dispatch:
+
+jobs:
+   build_image:
+      name: Build Vigilo image and push to ECR
+      runs-on: ubuntu-latest
+      permissions:
+        id-token: write
+        contents: read
+
+      steps:
+       - name: Checkout
+         uses: actions/checkout@v3
+
+       - name: Configure AWS credentials
+         uses: aws-actions/configure-aws-credentials@v2
+         with:
+            role-to-assume: ${{ env.DEPLOY_IAM_ROLE }}
```

Comment:
dev_block ⚡ I think we should use a repo secret instead of a repo variable for the iAM role arn. The variable can be logged as plain text and unlike the `data-pipelines` repo, this is a public repo.

Reply:
Updated in https://github.com/iFixit/vigilo/pull/18#discussion_r1416125742.

Reply:
Considering that our [iFixit/lighthouse-docker](https://github.com/iFixit/lighthouse-docker) repository is already public, and this new repository stems from it, I thought it would be reasonable to also make this one public. 

Also, this repository isn't closely tied to our core business logic, which makes it a suitable for open-sourcing, potentially helping others who might find this tool useful.